### PR TITLE
Add 'ml.reset_job' API and 'MlJob.blocked'

### PR DESCRIPTION
**Pull request by sethmlarson**
Closes https://github.com/elastic/elasticsearch-specification/issues/494

@jgowdyelastic can you take a look and verify this matches what you expect?
There don't seem to be any YAML tests for the `blocked` attribute (can we get one or more of these?)

### CODE REVIEW

File: output/typescript/types.ts [hunk 1]
```diff
@@ -10716,6 +10717,13 @@ export interface MlJob {
   system_annotations_retention_days?: long
 }
 
+export interface MlJobBlocked {
+  reason: MlJobBlockedReason
+  task_id: TaskId
```
Comment (jgowdyelastic) [inline output/typescript/types.ts:10722]: `task_id` should be optional

Comment (sethmlarson) [reply output/typescript/types.ts]: Done!

**Review by delvedor — APPROVED** [review]
LGTM

**Review by jgowdyelastic — APPROVED** [review]
LGTM

**Pull request merged**